src/Connector.php

```diff
@@ -4,28 +4,126 @@
 
-    public function __construct(LoopInterface $loop, Resolver $resolver)
+    public function __construct(LoopInterface $loop, array $options = array())
     {
-        $this->connector = new DnsConnector(new TcpConnector($loop), $resolver);
+        // apply default options if not explicitly given
+        $options += array(
+            'tcp' => true,
+            'tls' => true,
+            'unix' => true,
+
+            'dns' => true,
+            'timeout' => true,
+        );
+
+        if ($options['timeout'] === true) {
+            $options['timeout'] = (float)ini_get("default_socket_timeout");
+        }
+
+        if ($options['tcp'] instanceof ConnectorInterface) {
+            $tcp = $options['tcp'];
+        } else {
+            $tcp = new TcpConnector(
+                $loop,
+                is_array($options['tcp']) ? $options['tcp'] : array()
+            );
+        }
+
+        if ($options['dns'] !== false) {
+            if ($options['dns'] instanceof Resolver) {
+                $resolver = $options['dns'];
+            } else {
+                $factory = new Factory();
+                $resolver = $factory->create(
+                    $options['dns'] === true ? '8.8.8.8' : $options['dns'],
+                    $loop
+                );
+            }
+
+            $tcp = new DnsConnector($tcp, $resolver);
+        }
+
+        $connectors = array();
+
+        if ($options['tcp'] !== false) {
+            $options['tcp'] = $tcp;
+
+            if ($options['timeout'] !== false) {
+                $options['tcp'] = new TimeoutConnector(
+                    $options['tcp'],
+                    $options['timeout'],
+                    $loop
+                );
+            }
+
+            $connectors['tcp'] = $options['tcp'];
+        }
+
+        if ($options['tls'] !== false) {
+            if (!$options['tls'] instanceof ConnectorInterface) {
+                $options['tls'] = new SecureConnector(
+                    $tcp,
+                    $loop,
+                    is_array($options['tls']) ? $options['tls'] : array()
+                );
+            }
+
+            if ($options['timeout'] !== false) {
+                $options['tls'] = new TimeoutConnector(
+                    $options['tls'],
+                    $options['timeout'],
+                    $loop
+                );
+            }
+
+            $connectors['tls'] = $options['tls'];
+        }
+
+        if ($options['unix'] !== false) {
+            if (!$options['unix'] instanceof ConnectorInterface) {
+                $options['unix'] = new UnixConnector($loop);
+            }
+            $connectors['unix'] = $options['unix'];
+        }
+
+        $this->connectors = $connectors;
     }
 
     public function connect($uri)
     {
-        return $this->connector->connect($uri);
+        $scheme = 'tcp';
+        if (strpos($uri, '://') !== false) {
+            $scheme = (string)substr($uri, 0, strpos($uri, '://'));
+        }
```

Comment:
Suggestion: if no scheme is explicitely given and port is 443, use tls connector.